python/pyiceberg/expressions/__init__.py

```diff
@@ -40,36 +59,36 @@ def __invert__(self) -> BooleanExpression:
         """Transform the Expression into its negated version."""
 
 
-class Term(Generic[T], ABC):
+class Term(Generic[L], ABC):
     """A simple expression that evaluates to a value"""
 
 
 class Bound(ABC):
     """Represents a bound value expression"""
 
 
-class Unbound(Generic[B], ABC):
+class Unbound(ABC):
     """Represents an unbound value expression"""
 
+    @property
     @abstractmethod
-    def bind(self, schema: Schema, case_sensitive: bool = True) -> B:
```

Comment:
Minor: it seems valuable to continue to have a `bind` method on all unbound things.

Reply:
Agreed, added it back